### ISSUE TITLE
self hosted registry password

### ISSUE THREAD

**Issue by mgkimsal** [caprover/caprover#489]
created a self-hosted registry, but did not see any password generated, nor do I see any docs on a default password, or ways to change it.  i also can't just delete the self-hosted and try again because I need to add another registry and set it as default push.

is there a password set/change process for self-hosted registries that I missed?

**Comment by githubsaturn** [comment]
Since it's a managed registry, password is set by CapRover. And it's not shown to the end user as they are not supposed to be required to use it. It is only useful for debugging purposes. If you really need it, you can try running `api` command on CapRover CLI:

```
caprover ls
# make sure you're logged in to your instance. It should be in the list.

caprover api
# select your server
# for path, use /user/registries
# for method, use "GET"
# for API data JSON string, leave it empty.
# the output is like this:
{
  "registries": [
    {
      "id": "abcd-abcd-abcd-abcd-abcd",
      "registryDomain": "registry.root.domain.com:996",
      "registryImagePrefix": "captain",
      "registryUser": "captain",
      "registryPassword": "abcd-abcd-abcd-abcd-abcd",
      "registryType": "LOCAL_REG"
    }
  ],
  "defaultPushRegistryId": ""
}
```

You can see the `registryPassword` in the response above.



By the way, you cannot delete the registry which is set to as "default push". First you need to change the default push to another registry, or set it to `disabled`, then you're able to delete a registry.

**Comment by alexander-potemkin** [comment]
@githubsaturn , what's the workflow supposed to be?
I've configured local registry recently, but wasn't able to see / edit password, until I found that post. Password field was saying 'edit to see the password', but editing for local registry was not available.

**Comment by githubsaturn** [comment]
> what's the workflow supposed to be?

Workflow for what? Viewing the password? Just follow the steps in my [comment above](https://github.com/caprover/caprover/issues/489#issuecomment-524311421). Using `caprover api` command, you'd be able to get the password.

**Comment by alexander-potemkin** [comment]
@githubsaturn , yeah, it would just nice to have that reference at the web interface as well, otherwise it feels like broken functionality...

**Comment by githubsaturn** [comment]
Let's reopen this as a feature request for people to upvote.

**Comment by 01Vladimir10** [comment]
Alternatively, you can just check the network tab in your browser while being in the cluster tab:

![image](https://github.com/caprover/caprover/assets/40363360/2dbe078f-e115-45ad-b8bd-58f49b2de7bc)

**Comment by TylonHH** [comment]
> Alternatively, you can just check the network tab in your browser while being in the cluster tab:

I tried this, and got `{"status":1106,"description":"Auth token corrupted","data":{}}`
any chance to get/set one?
